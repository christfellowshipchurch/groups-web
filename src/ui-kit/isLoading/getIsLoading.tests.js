import React from 'react';
import renderer from 'react-test-renderer';

// import Providers from '../../Providers';

import { getIsLoading, IsLoadingContext } from '.';

describe('the getIsLoading HOC', () => {
  it('should return false from context', () => {
    const Context = getIsLoading(({ isLoading }) => isLoading.toString());

    const tree = renderer.create(
      <IsLoadingContext.Provider value={false}>
        <Context />
      </IsLoadingContext.Provider>
    );

    expect(tree).toMatchSnapshot();
  });
  it('should return true from context', () => {
    const Context = getIsLoading(({ isLoading }) => isLoading.toString());

    const tree = renderer.create(
      <IsLoadingContext.Provider value>
        <Context />
      </IsLoadingContext.Provider>
    );

    expect(tree).toMatchSnapshot();
  });
  it('should return the isLoading value (true) from the component', () => {
    const Context = getIsLoading(({ isLoading }) => isLoading.toString());

    const tree = renderer.create(
      <IsLoadingContext.Provider>
        <Context isLoading />
      </IsLoadingContext.Provider>
    );

    expect(tree).toMatchSnapshot();
  });
});
