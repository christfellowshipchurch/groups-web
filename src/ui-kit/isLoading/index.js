import React, { useContext } from 'react';

// Create our IsLoadingContext
const IsLoadingContext = React.createContext(false);

/* Safely checks for isLoading to make it easy to build components that respond to either props or
 * context for rendering placeholder/loading state. */
const getIsLoading = (Component) => (props) => {
  // does the component have an isLoading property?
  if (Object.hasOwnProperty.call(props, 'isLoading')) {
    return <Component {...props} />;
  }

  // No? That's fine we'll pull it from context
  const isLoading = useContext(IsLoadingContext);

  return <Component isLoading={isLoading} {...props} />;
};

// Allows a component to accept an `isLoading` prop to pass down into context for it's children.
const withIsLoading = (Component) => ({ isLoading, ...props }) => (
  <IsLoadingContext.Provider value={isLoading}>
    <Component {...props} />
  </IsLoadingContext.Provider>
);

export { getIsLoading, IsLoadingContext, withIsLoading };
