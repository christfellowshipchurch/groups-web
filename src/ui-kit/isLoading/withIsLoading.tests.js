import React from 'react';
import renderer from 'react-test-renderer';

import { getIsLoading, withIsLoading } from '.';

describe('the withIsLoading HOC', () => {
  it('should render', () => {
    const ParentComponentPassingConext = withIsLoading(
      ({ children }) => children
    );

    const tree = renderer.create(
      <ParentComponentPassingConext>
        <h1>Boom</h1>
      </ParentComponentPassingConext>
    );

    expect(tree).toMatchSnapshot();
  });
  it('should pass the isLoading prop (true) to context for children to consume', () => {
    const ChildComponentGettingContext = getIsLoading(({ isLoading }) =>
      isLoading.toString()
    );

    const ParentComponentPassingConext = withIsLoading(() => (
      <ChildComponentGettingContext />
    ));

    const tree = renderer.create(<ParentComponentPassingConext isLoading />);

    expect(tree).toMatchSnapshot();
  });
});
