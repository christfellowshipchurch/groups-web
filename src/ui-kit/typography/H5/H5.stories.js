import React from 'react';
import { storiesOf } from '@storybook/react';

import H5 from '.';

storiesOf('ui-kit/typography/H5', module)
  .add('default', () => (
    <H5>
      {
        '"God is most glorified in us when we are most satisfied in Him" ― John Piper'
      }
    </H5>
  ))
  .add('isLoading', () => (
    <H5 isLoading showLoadingAnimation>
      {
        '"God is most glorified in us when we are most satisfied in Him" ― John Piper'
      }
    </H5>
  ))
  .add('withMargins', () => <H5 withMargins>Padded Heading Big and Long</H5>)
  .add('withMargins + isLoading', () => (
    <H5 isLoading withMargins showLoadingAnimation>
      {
        '"God is most glorified in us when we are most satisfied in Him" ― John Piper'
      }
    </H5>
  ));
