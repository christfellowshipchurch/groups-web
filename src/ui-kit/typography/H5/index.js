import React from 'react';
import styled, { css } from 'styled-components';
import PropTypes from 'prop-types';

import Placeholder from '../../Placeholder';

const h5Style = css`
  h5,
  .H5 {
    color: ${({ theme }) => theme.colors.text.primary};
    font-family: ${({ theme }) => theme.typography.fontFamily};
    font-size: 0.875rem;
    font-weight: 500;
    line-height: ${({ theme }) => theme.helpers.verticalRhythm(0.875)};
    &.custom-placeholder {
      height: 0.875rem; /* Should match .h5font-size */
    }
  }
`;

const StyledH5 = styled.h5.attrs({
  className: 'ui-kit-typography-H5-StyledH5',
})`
  margin: ${({ padded, theme }) =>
    padded
      ? `${theme.helpers.verticalRhythm(
          0.5775
        )} 0 ${theme.helpers.verticalRhythm(0.4375)} 0`
      : 'inital'};
`;

const H5 = ({ children, isLoading, padded, ...props }) => (
  <Placeholder.Typography
    isLoading={isLoading}
    showLoadingAnimation
    typographicClassName={'H5'} // needs to match child components className
  >
    <StyledH5 padded={padded} {...props}>
      {children}
    </StyledH5>
  </Placeholder.Typography>
);

H5.defaultProps = {
  isLoading: false,
};

H5.propTypes = {
  children: PropTypes.node.isRequired,
  isLoading: PropTypes.bool,
  padded: PropTypes.bool,
};

H5.displayName = 'ui-kit.typography.H5';

export { H5 as default, h5Style };
