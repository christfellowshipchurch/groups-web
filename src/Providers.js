import React from 'react';
import { ThemeProvider } from 'styled-components';
import PropTypes from 'prop-types';

import { defaultTheme } from './ui-kit/theme';

const Providers = ({ children }) => (
  <ThemeProvider theme={defaultTheme}>{children}</ThemeProvider>
);

Providers.propTypes = {
  children: PropTypes.oneOfType([
    PropTypes.arrayOf(PropTypes.node),
    PropTypes.node,
  ]).isRequired,
};

export default Providers;
