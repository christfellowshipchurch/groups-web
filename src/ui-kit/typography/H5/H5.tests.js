import React from 'react';
import renderer from 'react-test-renderer';
import 'jest-styled-components';

import Providers from '../../../Providers';
import { IsLoadingContext } from '../../isLoading';

import H5 from '.';

describe('the H5 component', () => {
  it('should render', () => {
    const tree = renderer.create(
      <Providers>
        <H5>Default H5 text</H5>
      </Providers>
    );
    expect(tree).toMatchSnapshot();
  });
  it('should accept and render passed in styles', () => {
    const salmonText = { color: 'salmon' };
    const tree = renderer.create(
      <Providers>
        <H5 style={salmonText}>Salmon text</H5>
      </Providers>
    );
    expect(tree).toMatchSnapshot();
  });
  it('should render with vertical rhythm (withMargins)', () => {
    const tree = renderer.create(
      <Providers>
        <H5 withMargins>H5 text with vertical rhythm styled via margin.</H5>
      </Providers>
    );
    expect(tree).toMatchSnapshot();
  });
  it('should render a loading state', () => {
    const tree = renderer.create(
      <Providers>
        <H5 isLoading>Default H5 text</H5>
      </Providers>
    );
    expect(tree).toMatchSnapshot();
  });
  it('should render a loading state via context', () => {
    const tree = renderer.create(
      <Providers>
        <IsLoadingContext.Provider value>
          <H5>Default H5 text</H5>
        </IsLoadingContext.Provider>
      </Providers>
    );
    expect(tree).toMatchSnapshot();
  });
  it('should render a loading state with vertical rhythm (withMargins)', () => {
    const tree = renderer.create(
      <Providers>
        <H5 isLoading withMargins>
          Default H5 text
        </H5>
      </Providers>
    );
    expect(tree).toMatchSnapshot();
  });
});
