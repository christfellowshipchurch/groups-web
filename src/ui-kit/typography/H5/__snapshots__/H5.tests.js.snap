// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`the H5 component should accept and render passed in styles 1`] = `
<h5
  className="sc-1hocjyp-0 dhcvWr ui-kit-typography-H5-StyledH5"
  style={
    Object {
      "color": "salmon",
    }
  }
>
  Salmon text
</h5>
`;

exports[`the H5 component should render 1`] = `
<h5
  className="sc-1hocjyp-0 dhcvWr ui-kit-typography-H5-StyledH5"
>
  Default H5 text
</h5>
`;

exports[`the H5 component should render a loading state 1`] = `
<mark
  className="ra2rjz-0 gJMToV H5 show-loading-animation ui-kit.Placeholder.Line custom-placeholder"
  width="100%"
/>
`;

exports[`the H5 component should render with vertical rhythm (padded) 1`] = `
<h5
  className="sc-1hocjyp-0 egaYBw ui-kit-typography-H5-StyledH5"
>
  H5 text with vertical rhythm styled via margin.
</h5>
`;
